bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoVacuumHandler.java

```diff
@@ -468,6 +542,62 @@ public void onMessageReceived(MiIoSendCommand response) {
         }
     }
 
+    private void setCapabilities(JsonObject statusResponse) {
+        for (RobotCababilities capability : RobotCababilities.values()) {
+            if (statusResponse.has(capability.getStatusFieldName())) {
+                deviceCapabilities.putIfAbsent(capability, false);
+                logger.debug("Setting additional vacuum {}", capability);
+            }
+        }
+    }
+
+    private void createCapabilityChannels() {
+        ThingBuilder thingBuilder = editThing();
+        int cnt = 0;
+
+        for (Entry<RobotCababilities, Boolean> robotCapability : deviceCapabilities.entrySet()) {
+            RobotCababilities capability = robotCapability.getKey();
+            Boolean channelCreated = robotCapability.getValue();
+            boolean channelfound = false;
+            if (!channelCreated) {
+                for (Channel ch : thing.getChannels()) {
+                    logger.debug("Channel :{}, Channeltype UID: {}", ch.getUID(), ch.getChannelTypeUID());
+                    if (ch.getUID().getId().equalsIgnoreCase(capability.getChannel())) {
+                        logger.debug("Channel already available...skip creation of channel '{}'.",
+                                capability.getChannel());
+                        deviceCapabilities.replace(capability, true);
+                        channelfound = true;
+                        break;
+                    }
+                }
+                if (channelfound) {
+                    break;
+                }
```

Comment:
One of the few times that streams improves readability.
```suggestion
                if(thing.getChannels().stream()
                        .anyMatch(ch -> ch.getUID().getId().equalsIgnoreCase(capability.getChannel()))) {
                    logger.debug("Channel already available...skip creation of channel '{}'.",
                            capability.getChannel());
                    deviceCapabilities.replace(capability, true);
                    continue;
                }
```

Also, I think you intended to `continue` instead of `break` in your initial code.

Reply:
thanks.